common/views/components/ButtonSolid/ButtonSolidLink.js

```diff
@@ -0,0 +1,60 @@
+import {
+  SolidButtonInner,
+  ButtonIconWrapper,
+  SolidButton,
+} from './ButtonSolid';
+import { trackEvent } from '@weco/common/utils/ga';
+import Icon from '../Icon/Icon';
+import NextLink from 'next/link';
+import type ButtonSolidProps from './ButtonSolid';
+import type NextLinkType from '../../../model/next-link-type';
+
+type ButtonSolidLinkProps = {|
+  ...ButtonSolidProps,
+  link: NextLinkType | string,
+|};
+
+const ConditionalWrapper = ({ condition, wrapper, children }) =>
+  condition ? wrapper(children) : children;
+
+const ButtonSolidLink = ({
+  text,
+  link,
+  icon,
+  trackingEvent,
+  clickHandler,
+  ariaControls,
+  ariaExpanded,
+  disabled,
+  big,
+}: ButtonSolidLinkProps) => {
+  function handleClick(event) {
+    clickHandler && clickHandler(event);
+    trackingEvent && trackEvent(trackingEvent);
+  }
+  return (
+    <ConditionalWrapper
+      condition={link.as}
+      wrapper={children => <NextLink {...link}>{children}</NextLink>}
+    >
+      <SolidButton
+        aria-controls={ariaControls}
+        aria-expanded={ariaExpanded}
+        onClick={handleClick}
+        big={big}
+        href={link.as ? undefined : link}
+      >
+        <SolidButtonInner>
+          {icon && (
+            <ButtonIconWrapper>
+              <Icon name={icon} />
+            </ButtonIconWrapper>
+          )}
+          {text}
+        </SolidButtonInner>
+      </SolidButton>
+    </ConditionalWrapper>
+  );
+};
+
+export default ButtonSolidLink;
```

Comment:
Could we not have the `ButtonSolid` categorisation? I was getting caught up again by having the `styled` directory. We can always define how these get categories in cardigan, but from a deve perspective having 1 way saves time.

Reply:
I don't follow. Could you explain what you mean a bit more?

We'd been planning on making a button (and _probably_ an equivalent link-that-looks-like-a-button) for each of the variants listed in #5230

So I think we'd end up with `ButtonSolid`, `ButtonOutlined`, `ButtonInline` etc.

Reply:
Sorry, meant the folder structure is all.

Reply:
Ah, ok – could you outline your preference?

Reply:
`ButtonSolid/ButtonSolid.js` etc?

Although if thinking had progressed all good, I just thought we agreed that yonks ago (which is also reliant on my memory 💨 ).